cluster/gce/coreos/node.yaml

```diff
@@ -75,36 +12,36 @@ coreos:
         [Service]
         Type=oneshot
         RemainAfterExit=yes
-        ExecStart=/usr/bin/curl --fail --silent --show-error \
+        ExecStartPre=/usr/bin/curl --fail --silent --show-error \
         -H "X-Google-Metadata-Request: True" \
-        -o /etc/kube-env \
+        -o /etc/kube-env.yaml \
         http://metadata.google.internal/computeMetadata/v1/instance/attributes/kube-env
+        # Transform the yaml to env file.
+        ExecStartPre=/usr/bin/mv /etc/kube-env.yaml /etc/kube-env
+        ExecStart=/usr/bin/sed -i "s/: '/=/;s/'$//" /etc/kube-env
 
     - name: kubernetes-install-rkt.service
       command: start
       content: |
         [Unit]
-        Description=Fetch Rocket
+        Description=Fetch rkt
         Documentation=http://github.com/coreos/rkt
         Requires=network-online.target
         After=network-online.target
         [Service]
         Type=oneshot
+        RemainAfterExit=yes
         EnvironmentFile=/etc/kube-env
-        ExecStartPre=/usr/bin/rm -rf /opt/rkt
-        ExecStartPre=/usr/bin/mkdir -p /opt/rkt
         ExecStartPre=/usr/bin/mkdir -p /etc/rkt
-        ExecStartPre=/usr/bin/wget \
-        -O /opt/rkt/rkt-v${RKT_VERSION}.tar.gz \
-        https://github.com/coreos/rkt/releases/download/v${RKT_VERSION}/rkt-v${RKT_VERSION}.tar.gz
-        ExecStartPre=/usr/bin/tar xzvf /opt/rkt/rkt-v${RKT_VERSION}.tar.gz -C /opt --overwrite
-        ExecStart=/usr/bin/mv /opt/rkt-v${RKT_VERSION} /opt/rkt/rkt
+        ExecStartPre=/usr/bin/mkdir -p /opt/rkt/release
+        ExecStartPre=/usr/bin/curl --location --create-dirs --output /opt/rkt/release/rkt.tar.gz https://github.com/coreos/rkt/releases/download/v${RKT_VERSION}/rkt-v${RKT_VERSION}.tar.gz
+        ExecStart=/usr/bin/tar xf /opt/rkt/release/rkt.tar.gz -C /opt --overwrite
```

Comment:
nit: inconsistent in download location. In the master config rkt is downloaded to /opt/downloads VS. here /opt/rkt/release